lib/Container.js

```diff
@@ -63,12 +58,38 @@ Container.prototype.buildSetupTasks = Promise.promisify(function(done) {
   done(null, tasks);
 });
 
+Container.prototype._createDownloader = function(build, project){
+  // map provider type to downloader class name
+  var downloaders = {
+    'github': 'GithubDownloader',
+    'stash': 'StashDownloader',
+  }
+
+  var provider_type = project.provider.type
+  var clazz = downloaders[provider_type]
```

Comment:
Why not use a real word here like `plugin` rather than a slight WTF like `clazz`?